romsel_aktheme/arm9/source/common/bootstrapconfig.cpp

```diff
@@ -424,3 +456,435 @@ int BootstrapConfig::launch()
 		_configSavedHandler();
 	return loader.launch(argarray.size(), (const char **)&argarray[0], (_isHomebrew ? false : true));
 }
+
+std::string BootstrapConfig::getAPFix()
+{
+	remove("fat:/_nds/nds-bootstrap/apFix.ips");
+
+	if (_isHomebrew) {
+		return "";
+	}
+
+	bool ipsFound = false;
+	char ipsPath[256];
+	snprintf(ipsPath, sizeof(ipsPath), "%s:/_nds/TWiLightMenu/apfix/%s.ips", sdFound() ? "sd" : "fat", _fileName.c_str());
+	ipsFound = (access(ipsPath, F_OK) == 0);
+
+	if (!ipsFound) {
+		snprintf(ipsPath, sizeof(ipsPath), "%s:/_nds/TWiLightMenu/apfix/%s-%X.ips", sdFound() ? "sd" : "fat", _gametid.c_str(), _headerCrc16);
+		nocashWrite(ipsPath, sizeof(ipsPath));
+		ipsFound = (access(ipsPath, F_OK) == 0);
+	}
+
+	if (ipsFound) {
+		if (ms().secondaryDevice && sdFound()) {
+			mkdir("fat:/_nds", 0777);
+			mkdir("fat:/_nds/nds-bootstrap", 0777);
+			fcopy(ipsPath, "fat:/_nds/nds-bootstrap/apFix.ips");
+			return "fat:/_nds/nds-bootstrap/apFix.ips";
+		}
+		return std::string(ipsPath);
+	}
+
+	return "";
+}
+
+bool BootstrapConfig::checkCompatibility() {
+	if (!isDSiMode()) {
+		// TODO: If the list gets large enough, switch to bsearch().
+		for (unsigned int i = 0; i < sizeof(incompatibleGameListB4DS)/sizeof(incompatibleGameListB4DS[0]); i++) {
+			if (memcmp(_gametid.c_str(), incompatibleGameListB4DS[i], 3) == 0) {
+				return false;
+			}
+		}
+	} 
+	for (unsigned int i = 0; i < sizeof(incompatibleGameList)/sizeof(incompatibleGameList[0]); i++) {
+		if (memcmp(_gametid.c_str(), incompatibleGameList[i], 3) == 0) {
+			// Found match
+			return false;
+		}
+	}
+	return true;
+}
+
+APFixType BootstrapConfig::hasAPFix() {
+	nocashWrite(_apFix.c_str(), _apFix.length());
+	if (!_apFix.empty()) return APFixType::EHasIPS; // we have a known AP fix ready.
+	nocashWrite(_gametid.c_str(), 4);
+	// Check for SDK4-5 ROMs that don't have AP measures.
+	if ((_gametid.rfind("AZLJ", 0) == 0)	// Girls Mode (JAP version of Style Savvy)
```

Comment:
Why not make the AP list universal, like donorMap, saveMap & speedBumpExcludeMap?

Reply:
out of scope for this PR. plan is to make `bootstrapconfig`, `widescreenconfig`, etc. universal and get rid of existing boot code in other themes